changelog/2024-07-26T07_36_58+02_00_ghc_910_upgrade

```diff
@@ -0,0 +1,4 @@
+ADDED: Support for GHC 9.10.
+A word of caution: When the Clash compiler is compiled against GHC 9.10, it will
+currently only work reliably on 64-bit platforms. Compile the Clash compiler with
+GHC 9.8 or older if you are on a 32-bit platform.
```

Comment:
I guess we could add that to `BrokenGhcs` (even though its more like `BrokenClash` in this case :-)).

Reply:
Could you make an issue out of this and link it here?

Reply:
Tracked by: https://github.com/clash-lang/clash-compiler/issues/2776